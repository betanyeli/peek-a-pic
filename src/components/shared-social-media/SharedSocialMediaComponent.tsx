import { View, Linking, Alert } from 'react-native';
import React from 'react';
import styles from './SharedSocialMediaComponent.styles';
import {
  shareSocialMediaData,
  shareSocialMediaDataItem,
} from '../../data/sharedSocialMediaData';
import { IconButton, Text } from 'react-native-paper';

const SharedSocialMediaComponent = () => {
  const handleOpenUrl = async (url: string) => {
    const supported = await Linking.canOpenURL(url);
    if (supported) {
      await Linking.openURL(url);
    } else {
      Alert.alert(`ðŸ˜± Try again later to open this URL: ${url}`);
    }
  };

  const title = 'This pic rocks!';
  const subtitle =
    'Share with your friends or better...\nGive feedback to the creator ðŸ¥°';
  return (
    <>
      <View style={styles.titleContainer}>
        <Text style={[styles.color, styles.title]}>{title}</Text>
        <Text style={[styles.color, styles.subtitle]}>{subtitle}</Text>
      </View>
      <View style={styles.iconContainer}>
        {shareSocialMediaData.map(
          ({ id, icon, url }: shareSocialMediaDataItem) => {
            return (
              <IconButton
                testID="iconButton"
                key={id}
                mode="contained"
                icon={icon}
                onPress={() => {
                  handleOpenUrl(url);
                }}
              />
            );
          },
        )}
      </View>
    </>
  );
};

export default SharedSocialMediaComponent;
