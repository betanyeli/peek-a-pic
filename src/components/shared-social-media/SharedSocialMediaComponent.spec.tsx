import SharedSocialMediaComponent from "./SharedSocialMediaComponent";
import { render, waitFor } from '@testing-library/react-native';

jest.useFakeTimers()
describe('SharedSocialMediaComponent test', () => {
    it('Should be match with snapshot', () => {
        //Arrange
        const { toJSON } = render(<SharedSocialMediaComponent />);
        // Assert

        // Act
        waitFor(() => { expect(toJSON).toMatchSnapshot() });
    });

});