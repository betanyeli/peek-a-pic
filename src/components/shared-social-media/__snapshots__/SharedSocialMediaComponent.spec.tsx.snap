// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SharedSocialMediaComponent test Should be match with snapshot 1`] = `[Function]`;
